### PR TITLE
Calendar Component

### PR DESCRIPTION
**Pull request by Kimeiga**
## Calendar Component
- Access Calendar view by clicking calendar view in navbar
- shows interactive calendar of all events on month
- clicking on a calendar cell displays all events on that day in sidebar

## Need to fix:
- Event descriptions within each calendar cell include all events with that day number rather than all events *in current month and year* with that day number.
- The sidebar grows too wide when there's a long name with no spaces (fix via css, unrelated to calendar)
- Filter UI in sidebar does not filter calendar event data (need to write filtering functions in `event.service.ts`

### CODE REVIEW

**Comment by TrinaKat** [comment]
Design: https://www.figma.com/file/coG0RWW0XmNcAlH2oGYC7lTb/Mappening?node-id=0%3A1222
image

Current Implementation
image

File: nginx/conf/nginx.conf [hunk 1]
```diff
@@ -29,7 +29,11 @@ http {
         listen 443;
 
         if ($http_x_forwarded_proto = 'http') {
+<<<<<<< HEAD
```
Comment (TrinaKat) [inline nginx/conf/nginx.conf]: Pls fix merge conflicts

File: package.json [hunk 2]
```diff
@@ -6,23 +6,21 @@
...
-    "@angular/service-worker": "^7.0.0",
+    "@angular/animations": "^7.0.2",
+    "@angular/common": "^7.0.2",
+    "@angular/compiler": "^7.0.2",
+    "@angular/core": "^7.0.2",
+    "@angular/forms": "^7.0.2",
+    "@angular/http": "^7.0.2",
+    "@angular/platform-browser": "^7.0.2",
+    "@angular/platform-browser-dynamic": "^7.0.2",
+    "@angular/router": "^7.0.2",
```
Comment (TrinaKat) [inline package.json]: Still need service worker, make sure implementation works with `yarn`

File: baha.diff [hunk 3]
```diff
@@ -0,0 +1,12 @@
+diff --git a/src/app/event.service.ts b/src/app/event.service.ts
```
Comment (TrinaKat) [inline baha.diff]: What is this file for

File: angular.json [hunk 4]
```diff
@@ -18,8 +18,7 @@
             "polyfills": "src/polyfills.ts",
             "assets": [
               "src/assets",
-              "src/favicon.ico",
-              "src/manifest.json"
```
Comment (TrinaKat) [inline angular.json]: If we don't need manifest.json why do we add one in this PR

Comment (Kimeiga) [reply angular.json]: for some reason i dont see these lines in angular.json

Comment (Kimeiga) [reply angular.json]: wait we need manifest.json for pwa

File: package.json (hunk 2 above)
Comment (Kimeiga) [reply package.json]: having trouble with yarn:

`info There appears to be trouble with your network connection. Retrying...`

I think ryan had this problem too. i think its windows specific. will try on my linux distro

File: angular.json (hunk 4 above)
Comment (Kimeiga) [reply angular.json]: hmm might have been lost in the merge, i'll add the line back

File: angular.json [hunk 5]
```diff
@@ -80,8 +78,7 @@
             ],
             "assets": [
               "src/assets",
-              "src/favicon.ico",
-              "src/manifest.json"
```
Comment (TrinaKat) [inline angular.json]: here too

File: package.json [hunk 6]
```diff
@@ -6,23 +6,21 @@
     "ng": "ng",
     "start": "ng serve",
     "build": "ng build --prod",
-    "sw-build": "ng build --prod && node build/fix-sw",
```
Comment (TrinaKat) [inline package.json]: is this needed

Comment (Kimeiga) [reply package.json]: not sure; adding it back

File: package.json (hunk 2 above)
Comment (Kimeiga) [reply package.json]: finished

**Comment by Kimeiga** [comment]
BUG: event descriptions in calendar cells only show on reload

**Pull request merged**